code/game/objects/structures/barricade/barricade.dm

```diff
@@ -441,28 +442,36 @@
 			repair_value = repair_materials[validSheetType]
 			break
 
+	if(!istype(material, /obj/item/stack/sheet/))
+		to_chat(user, SPAN_WARNING("You'll need some adequate repair material in your other hand to patch up [src]!"))
+		return FALSE
+
 	if(repair_value == 0)
 		to_chat(user, SPAN_WARNING("You'll need some adequate repair material in your other hand to patch up [src]!"))
 		return FALSE
 
-	var/soundchannel = playsound(src, nailgun.repair_sound, 25, 1)
-	if(!do_after(user, nailgun.nailing_speed, INTERRUPT_ALL, BUSY_ICON_FRIENDLY, src))
-		playsound(src, null, channel = soundchannel)
+	if(istype(item,/obj/item/weapon/gun/smg/nailgun/compact/tactical) && material.amount < mat_tac) //If less than [mat_tac] material is held, cancel repair.
+		to_chat(user, SPAN_WARNING("You need atleast [mat_tac] [material.singular_name]\s to patch up [src]!"))
```

Comment:
"at least" - i also suspect there's a better solution than using \s here

Reply:
the \s ensures correct grammar for the material singular or plural form. Without it you end up with “4 metal sheet”.

Reply:
That looks a bit backwards to me but stack code does indeed does it, nevermind me